### ISSUE TITLE
Support package asset signing during build 

### ISSUE THREAD

**Issue by Stefan75** [NuGet/Home#12421]
### NuGet Product(s) Involved

MSBuild.exe

### The Elevator Pitch

Hello,

We want to ensure that all package asset files of all consumed packages are digital signed
so that the software supports environemnts with enabled Device-Guard (based on certifacates).

Regards,
Stefan

### Additional Context and Details

Idea:
- If package asset files are copied to the output folder we want to enrich the copy operation in a way that these files are automatically signed if necessary 
- This signing should happen before the actual target is created to ensure that the assets are always older than the created output target
- If multiple targets are sharing the output folder and using the same NuGet packages, the other targets should not replace the signed asset file in the output folder

Any suggestions?

**Comment by heng-liu** [comment]
Hi @Stefan75 , thanks for your suggestion!
You may check https://github.com/dotnet/sign and see if that's the feature you'd like to have.

//cc @dtivel

**Comment by Stefan75** [comment]
Hi @heng-liu , thanks for your reply.

If I understand the approach correctly, it seems that this tool signs in a late phase, after msbuild completion (e.g. the created msi).

I will discuss this approach with my colleagues. Until now, we thought more about signing as part of the build it self. 
We have also to thought about incremental builds. 

Thanks for the idea, any feedback to my initial idea?

Regards,
stefan

**Comment by heng-liu** [comment]
Hi @Stefan75 , I think signing asset files is out of scope for NuGet tooling. So I'll close this issue.

From [signing-tool-spec.md](https://github.com/dotnet/sign/blob/main/docs/signing-tool-spec.md), I learnt that:
`These files need to be signed "inside out" to ensure the proper sequence. `.
And here is an https://github.com/dotnet/sign/blob/main/docs/comparisons.md doc.
If you have more questions, could you raise them in https://github.com/dotnet/sign ? Thanks!
